### ISSUE TITLE
Package `emacs` allows any value for 'toolkit' when '~X'

### ISSUE THREAD

**Issue by TheQueasle** [LLNL/spack#2913]
Noticed this today while updating  #2884

The package logic raises an error if you supply `+X` and `toolkit` is something other than `gtk` and `athena`, but allows the install if `~X` and toolkit is anything.
```
[vagrant@sandbox ncbi-vdb]$ spack install emacs toolkit=sooperawesometoolkit
==> Installing emacs
==> ncurses is already installed in /home/vagrant/spack/opt/spack/linux-centos7-x86_64/gcc-4.8.5/ncurses-6.0-jfac4tqswyeppf5jphk2lxew7bpqx4z6
==> Using cached archive: /home/vagrant/spack/var/spack/cache/emacs/emacs-25.1.tar.gz
==> Already staged emacs-25.1-hmk2rtfrxmnay5tqbogsnbydymujctzp in /home/vagrant/spack/var/spack/stage/emacs-25.1-hmk2rtfrxmnay5tqbogsnbydymujctzp
==> Already patched emacs
==> Building emacs [AutotoolsPackage]
==> Executing phase : 'autoreconf'
==> Executing phase : 'configure'
==> Executing phase : 'build'
==> Executing phase : 'install'
==> Successfully installed emacs
  Fetch: 0.10s.  Build: 1m 0.90s.  Total: 1m 1.00s.
[+] /home/vagrant/spack/opt/spack/linux-centos7-x86_64/gcc-4.8.5/emacs-25.1-hmk2rtfrxmnay5tqbogsnbydymujctzp
```
I can work on a fix and a PR if this is indeed an issue.

**Comment by adamjstewart** [comment]
@TheQueasle This should be fixed when #2386 is merged and we can properly define a list of possible values for a multi-valued variant. For the time being, I wouldn't worry about it too much.

**Comment by TheQueasle** [comment]
:cool: Sounds good.
